src/Finder/RemovableRedirectFinder.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Setono\SyliusRedirectPlugin\Finder;
+
+use Doctrine\Common\Collections\ArrayCollection;
+use Doctrine\Common\Collections\Collection;
+use Setono\SyliusRedirectPlugin\Model\RedirectInterface;
+use Setono\SyliusRedirectPlugin\Repository\RedirectRepositoryInterface;
+use Setono\SyliusRedirectPlugin\Resolver\RedirectionPathResolverInterface;
+
+final class RemovableRedirectFinder implements RemovableRedirectFinderInterface
+{
+    /** @var RedirectRepositoryInterface */
+    private $redirectRepository;
+
+    /** @var RedirectionPathResolverInterface */
+    private $redirectionPathResolver;
+
+    public function __construct(RedirectRepositoryInterface $redirectRepository, RedirectionPathResolverInterface $redirectionPathResolver)
+    {
+        $this->redirectRepository = $redirectRepository;
+        $this->redirectionPathResolver = $redirectionPathResolver;
+    }
+
+    public function findNextRedirect(RedirectInterface $redirect): Collection
```

Comment:
I am not completely sure what this method does. Can you explain it? :)

Reply:
I'v add comment to the function. Is it clearer ?

Reply:
Can you add an example to the phpdoc?

Reply:
And maybe the explanation should go into the interface instead of the concrete class